src/middlewared/middlewared/etc_files/local/ssh/sshd_config.mako

```diff
@@ -1,18 +1,27 @@
 <%
 	import os
 	import ipaddress
+	import stat
+        from middlewared.utils import filter_list
 
-	ssh_config = middleware.call_sync('ssh.config')
-	if not os.path.exists('/root/.ssh'):
-		os.makedirs('/root/.ssh')
+	ssh_config = render_ctx['ssh.config']
+
+	os.makedirs('/root/.ssh', mode=0o700, exist_ok=True)
+	for p in ['/root/.ssh', '/root']:
+		st = os.stat(p)
+		if stat.S_IMODE(st.st_mode) != 0o700:
+			middleware.logger.debug("%s: adjusting permissions to 0o700", p)
+			os.chmod(p, 0o700)
+		if st.st_uid != 0 or st.st_gid != 0:
```

Comment:
Are you meaning to log when changing ownership since you're logging when changing mode?